apiary-metastore-listener/src/main/java/com/expedia/apiary/extensions/metastore/listener/ApiarySnsListener.java

```diff
@@ -148,23 +149,28 @@ private void publishEvent(
     }
     if (partition != null) {
       JSONArray partitionValuesArray = new JSONArray(partition.getValues());
-      json.put("partition", partitionValuesArray);
+      JSONArray partitionKeysArray = new JSONArray(
+          table.getPartitionKeys().stream().map(f -> f.getName()).collect(Collectors.toList()));
+
+      json.put("partitionKeys", partitionKeysArray);
```

Comment:
Where are the types?

Reply:
I reverted all the changes as the order was not being preserved for the keys. I will explore some other method of achieving this and will update here.